src/main/java/com/garannodou/role/controller/GoogleAuthRequest.java

```diff
@@ -0,0 +1,49 @@
+package com.garannodou.role.controller;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import javax.validation.constraints.NotNull;
+
+// TODO: Add concrete types and validations to each field, once the content is clearer.
+@Getter
+@Setter
+public class GoogleAuthRequest {
+
+    @NotNull
+    private GoogleBasicProfile googleBasicProfile;
+    @NotNull
+    private GoogleAuthResponse googleAuthResponse;
+
+    public GoogleAuthRequest() {
+    }
+
+    @Getter
+    @Setter
+    static class GoogleBasicProfile {
```

Comment:
"static" may not be needed?

Reply:
"static" not needed. GoogleBasicProfile and GoogleAuth only make sense to exist within GoogleAuthRequest.